include/SparseTriplet.hpp

```diff
@@ -14,5 +14,28 @@ namespace blitzdg {
         index_type* row; /**< Pointer to array of row indices. */
         index_type* col; /**< Pointer to array of column indices. */
         real_type* val;  /**< Pointer to array of elements. */
+
+        /**
+         * Constructor.
+         * Creates a sparse triplet matrix with space for nz_ elements.
+         * @param nz_ Number of nonzero elements.
+         * @note If nz_ <= 0, then the empty matrix is created. The number of 
+         * nonzeros is set to zero and the pointers are set to null.
+         */
+        explicit SparseTriplet(index_type nz_ = 0)
```

Comment:
I think I like having this guy manage his own memory like this. It jives better with the rest of the classes.

Reply:
I think it's a bit safter this way, especially with the destructor. What you need to be careful of is the following situation. Suppose you have a SparseTriplet named triplet that is pointing to some memory and you do something like triplet.row = new index_type[nz]; Unless you freed the memory that triplet.row was pointing to before doing the assignment, then that memory is lost and cannot be deallocated, i.e., you have a memory leak.